modules/page-memory/src/main/java/org/apache/ignite/internal/pagememory/persistence/checkpoint/CheckpointMetricsTracker.java

```diff
@@ -191,78 +195,78 @@ public void onSplitAndSortCheckpointPagesStart() {
     }
 
     /**
      * Returns total checkpoint duration.
      *
      * <p>Not thread safe.
      */
-    public long totalDuration() {
-        return checkpointEndTimestamp - checkpointStartTimestamp;
+    public long totalDuration(TimeUnit timeUnit) {
+        return timeUnit.convert(endNanos - startNanos, NANOSECONDS);
     }
 
     /**
      * Returns checkpoint write lock wait duration in mills.
      *
      * <p>Not thread safe.
      */
-    public long writeLockWaitDuration() {
-        return checkpointOnMarkCheckpointBeginStartTimestamp - checkpointWriteLockWaitStartTimestamp;
+    public long writeLockWaitDuration(TimeUnit timeUnit) {
+        return timeUnit.convert(onMarkCheckpointBeginStartNanos - writeLockWaitStartNanos, NANOSECONDS);
     }
 
     /**
      * Returns checkpoint action before taken write lock duration in mills.
      *
      * <p>Not thread safe.
      */
-    public long beforeWriteLockDuration() {
-        return checkpointWriteLockWaitStartTimestamp - checkpointStartTimestamp;
+    public long beforeWriteLockDuration(TimeUnit timeUnit) {
+        return timeUnit.convert(writeLockWaitStartNanos - startNanos, NANOSECONDS);
     }
 
     /**
      * Returns execution all {@link CheckpointListener#onMarkCheckpointBegin} under write lock duration in mills.
      *
      * <p>Not thread safe.
      */
-    public long onMarkCheckpointBeginDuration() {
-        return checkpointOnMarkCheckpointBeginEndTimestamp - checkpointOnMarkCheckpointBeginStartTimestamp;
+    public long onMarkCheckpointBeginDuration(TimeUnit timeUnit) {
+        return timeUnit.convert(onMarkCheckpointBeginEndNanos - onMarkCheckpointBeginStartNanos, NANOSECONDS);
     }
 
     /**
      * Returns checkpoint write lock hold duration in mills.
      *
      * <p>Not thread safe.
      */
-    public long writeLockHoldDuration() {
-        return checkpointWriteLockReleaseTimestamp - checkpointOnMarkCheckpointBeginStartTimestamp;
+    public long writeLockHoldDuration(TimeUnit timeUnit) {
+        return timeUnit.convert(writeLockReleaseNanos - onMarkCheckpointBeginStartNanos, NANOSECONDS);
     }
 
     /**
      * Returns pages write duration in mills.
      *
      * <p>Not thread safe.
      */
-    public long pagesWriteDuration() {
-        return checkpointFsyncStartTimestamp - checkpointPagesWriteStartTimestamp;
+    public long pagesWriteDuration(TimeUnit timeUnit) {
+        return timeUnit.convert(fsyncStartNanos - pagesWriteStartNanos, NANOSECONDS);
     }
 
     /**
      * Returns checkpoint fsync duration in mills.
      *
      * <p>Not thread safe.
      */
-    public long fsyncDuration() {
-        return checkpointEndTimestamp - checkpointFsyncStartTimestamp;
+    public long fsyncDuration(TimeUnit timeUnit) {
+        return timeUnit.convert(endNanos - fsyncStartNanos, NANOSECONDS);
     }
 
     /**
      * Returns duration of splitting and sorting checkpoint pages in mills.
```

Comment:
```suggestion
     * Returns duration of splitting and sorting checkpoint pages in the given time unit.
```
(the same for other methods)